tests/System.Slices.Tests/MemoryTests.cs

```diff
@@ -30,25 +31,72 @@ public void SimpleTestS()
         [Fact]
         public void ArrayMemoryLifetime()
         {
-            Memory<byte> copyStoredForLater;
-            using (var owner = new OwnedArray<byte>(1024)) {
-                Memory<byte> memory = owner.Memory;
-                Memory<byte> memorySlice = memory.Slice(10);
-                copyStoredForLater = memorySlice;
-                using (memorySlice.Reserve()) { // increments the "outstanding span" refcount
-                    Assert.Throws<InvalidOperationException>(() => { // memory is reserved; cannot dispose
-                        owner.Dispose();
-                    });
-                    Span<byte> span = memorySlice.Span;
-                    span[0] = 255;
-                } // releases the refcount
-            }
+            Memory<byte> copyStoredForLater = default(Memory<byte>);
+            Assert.Throws<InvalidOperationException>(() => {
```

Comment:
The standard pattern is to allow multiple calls to Dispose. From Dispose documentation: `If an object's Dispose method is called more than once, the object must ignore all calls after the first one`

I think Dispose on OwnedMemory should follow it for consistency.